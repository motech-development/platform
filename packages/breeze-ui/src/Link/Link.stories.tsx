import { withA11y } from '@storybook/addon-a11y';
import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import BaseStyles from '../BaseStyles/BaseStyles';
import Card from '../Card/Card';
import Link from './Link';

const stories = storiesOf('Link', module);

stories.addDecorator(withA11y);
stories.addDecorator(withKnobs);

stories.add('Basic link', () => (
  <MemoryRouter>
    <BaseStyles />

    <Card>
      <Link to={text('To', '/home')}>{text('Text', 'This is a link')}</Link>
    </Card>
  </MemoryRouter>
));
