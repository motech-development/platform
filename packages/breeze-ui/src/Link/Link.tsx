import { FC, memo } from 'react';
import { Link as RouterLink, LinkProps } from 'react-router-dom';
import BaseLink from '../BaseLink/BaseLink';

const Link: FC<LinkProps> = (props) => (
  // eslint-disable-next-line react/jsx-props-no-spreading
  <BaseLink as={RouterLink} {...props} />
);

export default memo(Link);
