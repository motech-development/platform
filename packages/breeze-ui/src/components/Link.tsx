import { FC } from 'react';

export interface ILinkProps {
  to: string;
}

const Link: FC<ILinkProps> = ({ to }) => (
  <button
    type="button"
    onClick={() => {
      window.location.href = to;
    }}
  >
    Test
  </button>
);

export default Link;
