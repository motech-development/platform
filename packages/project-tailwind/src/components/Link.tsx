import { Slot } from '@radix-ui/react-slot';
import { ComponentPropsWithoutRef, forwardRef } from 'react';
import {
  TTheme,
  TSizing,
  Themes,
  Sizing,
  useTailwind,
} from '../utilities/tailwind';
import { ISlot } from './Slot';

/** Default component element */
const DEFAULT_ELEMENT = 'a';

/** Link component props */
export interface ILinkProps extends ISlot, ComponentPropsWithoutRef<'a'> {
  /** Renders component as a block element */
  block?: boolean;

  /** Component theme */
  theme?: TTheme;

  /** Component size */
  size?: TSizing;
}

/**
 * Flexible anchor element
 *
 * @param props - Link component props
 *
 * @returns Link component
 */
export const Link = forwardRef<HTMLAnchorElement, ILinkProps>(
  (
    {
      asChild,
      block = false,
      className,
      theme = Themes.PRIMARY,
      size = Sizing.MD,
      ...rest
    },
    ref,
  ) => {
    const { createStyles } = useTailwind(theme, size);

    const Comp = asChild ? Slot : DEFAULT_ELEMENT;

    const linkStyles = createStyles({
      classNames: [
        'appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2',
        {
          'block w-full text-center': block,
          'inline-block': !block,
        },
        className,
      ],
      sizing: {
        lg: ['text-lg'],
        md: ['text-base'],
        none: ['text-base'],
        sm: ['text-sm'],
      },
      theme: {
        danger: ['text-red-600 hover:text-red-700 focus:ring-red-500'],
        primary: ['text-blue-600 hover:text-blue-700 focus:ring-blue-500'],
        secondary: ['text-gray-600 hover:text-gray-700 focus:ring-gray-500'],
        success: ['text-green-600 hover:text-green-700 focus:ring-green-500'],
        warning: [
          'text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500',
        ],
      },
    });

    return <Comp className={linkStyles} ref={ref} {...rest} />;
  },
);
