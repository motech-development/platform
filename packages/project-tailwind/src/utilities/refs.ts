import { LegacyRef, MutableRefObject, useMemo, useRef } from 'react';

/**
 * Check if ref is a mutable ref object
 *
 * @param ref - Reference object or function
 *
 * @returns If ref is a mutable ref object
 */
function isMutableRefObject<T>(
  ref: MutableRefObject<T> | LegacyRef<T>,
): ref is MutableRefObject<T | null> {
  return ref != null;
}

/** useMergeRefs arguments */
type TUseMergeRefsInput<T> = Array<MutableRefObject<T> | LegacyRef<T>>;

/**
 * Combines multiple refs into a single memoized callback instance
 *
 * @param refs - Reference object or function
 *
 * @returns Combined ref object
 */
export default function useMergeRefs<T>(...refs: TUseMergeRefsInput<T>) {
  const targetRef = useRef<T>(null);

  return useMemo(() => {
    if (refs.every((ref) => ref == null)) {
      return null;
    }

    refs.forEach((ref) => {
      if (typeof ref === 'function') {
        ref(targetRef.current);
      } else if (isMutableRefObject(ref)) {
        // eslint-disable-next-line no-param-reassign
        ref.current = targetRef.current;
      }
    });

    return targetRef;
  }, [refs]);
}
