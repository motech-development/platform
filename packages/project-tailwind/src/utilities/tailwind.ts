import clsx, { ClassValue } from 'clsx';

/** Alignment options */
export enum Alignment {
  /** Centre */
  CENTRE = 'centre',
  /** Left */
  LEFT = 'left',
  /** Right */
  RIGHT = 'right',
}

/** Alignment type */
export type TAlignment = `${Alignment}`;

/** Sizing options */
export enum Sizing {
  /** Large */
  LG = 'lg',
  /** Medium */
  MD = 'md',
  /** None */
  NONE = 'none',
  /** Small */
  SM = 'sm',
}

/** Sizing type */
export type TSizing = `${Sizing}`;

/** Theming options */
export enum Themes {
  /** Danger */
  DANGER = 'danger',
  /** Primary */
  PRIMARY = 'primary',
  /** Secondary */
  SECONDARY = 'secondary',
  /** Success */
  SUCCESS = 'success',
  /** Warning */
  WARNING = 'warning',
}

/** Theme type */
export type TTheme = `${Themes}`;

/** Class names options */
type TCreateStylesClassNames = {
  /** Universal classes to apply */
  classNames?: ClassValue[];
  /** Sizing specific classes */
  sizing?: {
    [name in Sizing]?: ClassValue[];
  };
  /** Theme specific classes */
  theme?: {
    [name in TTheme]?: ClassValue[];
  };
};

/**
 * Tailwind hook
 *
 * @param theme - The theme to apply
 * @param sizing - Sizing to apply
 *
 * @returns Tailwind helper utilities
 */
export function useTailwind(theme: TTheme, sizing: TSizing) {
  /**
   * Creates css styles to apply
   *
   * @param classNames - Classes to be applied
   *
   * @returns Class names
   */
  function createStyles(classNames: TCreateStylesClassNames) {
    let finalStyles: ClassValue[] = [];

    if (classNames.classNames) {
      finalStyles = [...classNames.classNames];
    }

    if (sizing && classNames.sizing) {
      const styles = classNames.sizing[sizing];

      if (!styles) {
        throw new Error('No classes defined for this sizing.');
      }

      finalStyles = [...finalStyles, ...styles];
    }

    if (theme && classNames.theme) {
      const styles = classNames.theme[theme];

      if (!styles) {
        throw new Error('No classes defined for this theme.');
      }

      finalStyles = [...finalStyles, ...styles];
    }

    return clsx(finalStyles);
  }

  /**
   * Creates text alignment classes
   *
   * @param alignment - Text alignment
   *
   * @returns Class names
   */
  function createTextAlignmentStyles(alignment: TAlignment) {
    switch (alignment) {
      case Alignment.CENTRE:
        return 'text-center';
      case Alignment.RIGHT:
        return 'text-right';
      case Alignment.LEFT:
      default:
        return 'text-left';
    }
  }

  return {
    createStyles,
    createTextAlignmentStyles,
  };
}
