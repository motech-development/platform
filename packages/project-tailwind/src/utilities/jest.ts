/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderResult } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { UserEvent } from '@testing-library/user-event/dist/types/setup/setup';
import { ReactElement } from 'react';
import { Sizing, Themes } from './tailwind';

export const themes = [
  {
    theme: Themes.DANGER,
  },
  {
    theme: Themes.PRIMARY,
  },
  {
    theme: Themes.SECONDARY,
  },
  {
    theme: Themes.SUCCESS,
  },
  {
    theme: Themes.WARNING,
  },
];

export const sizing = [
  {
    size: Sizing.LG,
  },
  {
    size: Sizing.MD,
  },
  {
    size: Sizing.NONE,
  },
  {
    size: Sizing.SM,
  },
];

type TSetup = RenderResult & {
  user: UserEvent;
};

/**
 * Combines Testing Library render function with user events setup
 *
 * @param ui - React element to test
 *
 * @returns Rendered component and test user
 */
export function setup(ui: ReactElement): TSetup {
  return {
    user: userEvent.setup({
      delay: null,
    }),
    ...render(ui),
  };
}
