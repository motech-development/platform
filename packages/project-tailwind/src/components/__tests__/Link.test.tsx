import { setup, sizing, themes } from '../../utilities/jest';
import { Link } from '../Link';

const elements = [
  {
    asChild: false,
    attrs: {
      href: 'https://motech-development.github.io/platform/',
    } as const,
    element: 'a',
    render: <>Hello, world</>,
  },
  {
    asChild: true,
    element: 'button',
    render: <button type="button">Hello, world</button>,
  },
];

describe('Link', () => {
  describe.each(elements)(
    'as element "$element"',
    ({ asChild, attrs, render }) => {
      describe.each(themes)('$theme', ({ theme }) => {
        it('should render the correct output when block element', () => {
          const { asFragment } = setup(
            <Link asChild={asChild} block theme={theme} {...attrs}>
              {render}
            </Link>,
          );

          expect(asFragment()).toMatchSnapshot();
        });

        it.each(sizing)(
          'should render the correct output when size is $size',
          ({ size }) => {
            const { asFragment } = setup(
              <Link asChild={asChild} size={size} theme={theme} {...attrs}>
                {render}
              </Link>,
            );

            expect(asFragment()).toMatchSnapshot();
          },
        );
      });
    },
  );
});
