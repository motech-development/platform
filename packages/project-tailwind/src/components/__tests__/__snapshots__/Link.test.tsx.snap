// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link as element "a" danger should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-red-600 hover:text-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" danger should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-red-600 hover:text-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" danger should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-red-600 hover:text-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" danger should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-red-600 hover:text-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" danger should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-red-600 hover:text-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" primary should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" primary should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" primary should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" primary should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" primary should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" secondary should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" secondary should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" secondary should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" secondary should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" secondary should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" success should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-green-600 hover:text-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" success should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-green-600 hover:text-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" success should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-green-600 hover:text-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" success should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-green-600 hover:text-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" success should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-green-600 hover:text-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" warning should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" warning should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" warning should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" warning should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "a" warning should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Hello, world
  </a>
</DocumentFragment>
`;

exports[`Link as element "button" danger should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-red-600 hover:text-red-700 focus:ring-red-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" danger should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-red-600 hover:text-red-700 focus:ring-red-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" danger should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-red-600 hover:text-red-700 focus:ring-red-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" danger should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-red-600 hover:text-red-700 focus:ring-red-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" danger should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-red-600 hover:text-red-700 focus:ring-red-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" primary should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" primary should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" primary should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" primary should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" primary should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-blue-600 hover:text-blue-700 focus:ring-blue-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" secondary should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" secondary should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" secondary should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" secondary should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" secondary should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-gray-600 hover:text-gray-700 focus:ring-gray-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" success should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-green-600 hover:text-green-700 focus:ring-green-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" success should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-green-600 hover:text-green-700 focus:ring-green-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" success should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-green-600 hover:text-green-700 focus:ring-green-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" success should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-green-600 hover:text-green-700 focus:ring-green-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" success should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-green-600 hover:text-green-700 focus:ring-green-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" warning should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 block w-full text-center text-base text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" warning should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-lg text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" warning should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" warning should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-base text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;

exports[`Link as element "button" warning should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="appearance-none bg-none border-0 font-semibold underline hover:no-underline focus:outline-none focus:ring-2 focus:ring-offset-2 inline-block text-sm text-yellow-600 hover:text-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Hello, world
  </button>
</DocumentFragment>
`;
